src/java.base/share/native/launcher/main.c

```diff
@@ -36,7 +36,11 @@
 
 #ifdef UNDEFINED_BEHAVIOR_SANITIZER
 // Override weak symbol exposed by UBSan to override default options. This is called by UBSan
-// extremely early during library loading, before main is called.
+// extremely early during library loading, before main is called. We need to override the default
+// options because by default UBSan only prints a warning for each occurrence. We want jtreg tests
+// to fail when undefined behavior is encountered. We also want a full stack trace for the offending
```

Comment:
If this is primarily for tests then can't we set the env-var in the test Makefile?

Reply:
Primarily, but its not a requirement. We should also be able to invoke `java` as is. The environment variables should be used to force specific behavior for a single invocation. Otherwise, if one forgets to pass the environment variables, things may not go as expected.

Reply:
It really comes down to who we expect to do these ubsan builds and how we expect them to be used. If this is something explicit then using the env-var seems fine to me. I'm not sure why it needs to be enabled by default through this source modification.

I'd like to hear other opinions on this.

Reply:
Sure. It should be noted that whatever we do here for default options will likely have to be repeated for LSan and ASan eventually. So it should be repeatable and easy to maintain. IMO we should enforce strict defaults and then let manual invocations make it less strict via environment variables.